services/web/server/src/simcore_service_webserver/projects/_folders_db.py

```diff
@@ -100,13 +103,80 @@ async def delete_project_to_folder(
         )
 
 
+### AsyncPg
+
+
 async def delete_all_project_to_folder_by_project_id(
     app: web.Application,
+    connection: AsyncConnection | None = None,
+    *,
     project_id: ProjectID,
 ) -> None:
-    async with get_database_engine(app).acquire() as conn:
-        await conn.execute(
+    async with transaction_context(get_asyncpg_engine(app), connection) as conn:
+        await conn.stream(
             projects_to_folders.delete().where(
                 projects_to_folders.c.project_uuid == f"{project_id}"
             )
         )
+
+
+async def update_project_to_folder(
+    app: web.Application,
+    connection: AsyncConnection | None = None,
+    *,
+    folders_id_or_ids: FolderID | set[FolderID],
+    # updatable columns
+    user_id: UserID | None | UnSet = _unset,
+) -> None:
+    """
+    Batch/single patch of project to folders
+    """
+    # NOTE: exclude unset can also be done using a pydantic model and dict(exclude_unset=True)
+    updated = as_dict_exclude_unset(
+        user_id=user_id,
+    )
+
+    query = projects_to_folders.update().values(modified=func.now(), **updated)
+
+    if isinstance(folders_id_or_ids, set):
+        # batch-update
+        query = query.where(
+            projects_to_folders.c.folder_id.in_(list(folders_id_or_ids))
+        )
+    else:
+        # single-update
+        query = query.where(projects_to_folders.c.folder_id == folders_id_or_ids)
+
+    async with transaction_context(get_asyncpg_engine(app), connection) as conn:
+        await conn.stream(query)
+
+
+async def delete_all_project_to_folder_by_project_ids_not_in_folder_ids(
+    app: web.Application,
+    connection: AsyncConnection | None = None,
+    *,
+    project_id_or_ids: ProjectID | set[ProjectID],
+    # Optional filter
+    not_in_folder_ids: set[FolderID],
+) -> None:
+    query = projects_to_folders.delete()
+
+    if isinstance(project_id_or_ids, set):
+        # batch-delete
+        query = query.where(
```

Comment:
THOUGHT: this part of the query is applicable to any batch operation we want, not only delete. E.g. it would be very easy to reuse this and have many of the batch alternatives that we pointed out in the other part of the code :-)

Reply:
yes you are right :+1: